utils/puppeteerScrapper.js

```diff
@@ -1,151 +1,90 @@
 const puppeteer = require('puppeteer'); // v13.0.0 or later
 
-const getWaifu = async () => {
-  const waifu = {
-    name: '',
-    imageUrl: '',
-    bioUrl: '',
-    extract: '',
-    appearsIn: '',
+const nameSelector = 'h1';
+const imageSelector = '#widget-waifu-of-the-day > a > img';
+const popularitySelector = '#popularity-rank';
+const ageSelector = '#age';
+const husbandoSelector = '#waifu-classification';
+const appearsSelector = '#waifu-core-information > div > div > a';
+
+const waifuURL = "https://mywaifulist.moe/"
+const randomUrl = "https://mywaifulist.moe/random"
+
+const getWaifu = async (retry = false) => {
```

Comment:
Initializing to a falsy value it's not needed unless you're going to check the type which it´s not the case here. This may indicate a type checking which may be slighlty confusing, please remove it.